[ci-visibility] Link RUM and CI Visibility (#12977)
DataDog/documentation: content/en/continuous_integration/guides/ci_visibility_rum_integration.md

---

> title: CI Visibility - RUM integration

Reviewer: ```suggestion
title: Instrumenting your browser tests with RUM
```

Maybe is a better title?

kind: guide
aliases: /continuous_integration/guides/ci_visibility_rum_integration/
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
{{< /site-region >}}

### Browser tests and RUM

> If your page under test is instrumented using [RUM][1], your `cypress` test results and their generated RUM browser sessions and session replays will be automatically linked. A new **Browser Sessions** tab appears in the browser test's test detail:

Reviewer 2: `If your page under test` 

Not sure exactly what this means

Author: this is the page you're running your browser tests against. See usage in cypress docs for example: https://docs.cypress.io/faq/questions/using-cypress-faq#Can-Cypress-be-used-for-performance-testing

> If your page under test is instrumented using [RUM][1], your `cypress` test results and their generated RUM browser sessions and session replays will be automatically linked. A new **Browser Sessions** tab appears in the browser test's test detail:

Reviewer: ```suggestion
If you use Cypress for your browser tests and your application under test is instrumented using [RUM][1], your test results and their generated RUM browser sessions and session replays will be automatically linked. A new **Browser Sessions** tab appears in the test detail side panel:
```

{{< img src="ci/ci-browser-session-tab.png" alt="Browser Session Tab in Test Detail" style="width:100%;">}}

This RUM session has all the data that [RUM normally collects][2], so it lets you debug potential issues in your browser test, such as unexpected errors:

{{< img src="ci/ci-browser-session-tab-errors.png" alt="Browser Session Tab Errors in Test Detail" style="width:100%;">}}

### Compatibility

> The CI Visibility - RUM integration is only available from these version of `dd-trace-js` and `browser-sdk`:

Reviewer 2: ```suggestion
The CI Visibility - RUM integration is only available for these versions of `dd-trace-js` and `browser-sdk`:
```

* `dd-trace-js>=1.7.0`

> * `browser-sdk>=3.11.0`

Reviewer: 1. Should we move this to the top, to be consistent with all the other pages?
2. Should we add here the testing frameworks we support (only Cypress) and its versions? The guide is about CI Visibility - RUM integration, but in reality we only support Cypress for now, so we have to be explicit about that

Author: that makes sense 👍

Author: done in https://github.com/DataDog/documentation/pull/12977/commits/e2e58c55b4ac7e88dcaa7d97a35763bbe286356e

Reviewer 3: I agree about adding compatibility to the top, so people know if it applies to them

---
title: Instrumenting your browser tests with RUM
kind: guide

> aliases: /continuous_integration/guides/ci_visibility_rum_integration/

Reviewer: ```suggestion
aliases: /continuous_integration/guides/rum_integration/
```

---

{{< site-region region="gov" >}}
<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
{{< /site-region >}}

### Compatibility

> The CI Visibility - RUM integration is only available for these versions of `cypress`, `dd-trace` and `browser-sdk`:

Reviewer: ```suggestion
The CI Visibility - RUM integration is only available for these versions of `cypress`, `dd-trace-js` and `browser-sdk`:
```

* `cypress` >= 6.7.0

> * `dd-trace` >= 1.7.0

Reviewer: ```suggestion
* `dd-trace-js` >= 1.7.0
```

* `browser-sdk` >= 3.11.0

**Note**: Only `cypress` is supported for this integration at the moment.

### Browser tests and RUM

If you use Cypress for your browser tests and your application under test is instrumented using [RUM][1], your test results and their generated RUM browser sessions and session replays will be automatically linked. A new **Browser Sessions** tab appears in the test detail side panel:

{{< img src="ci/ci-browser-session-tab.png" alt="Browser Session Tab in Test Detail" style="width:100%;">}}

This RUM session has all the data that [RUM normally collects][2], so it lets you debug potential issues in your browser test, such as unexpected errors:

{{< img src="ci/ci-browser-session-tab-errors.png" alt="Browser Session Tab Errors in Test Detail" style="width:100%;">}}

> 

Reviewer: Maybe add something about session replay? We can also add it later
